decompiler/frontend/binaryninja/handlers/constants.py

```diff
@@ -76,69 +40,20 @@ def lift_constant_pointer(self, pointer: mediumlevelil.MediumLevelILConstPtr, **
         if function := view.get_function_at(pointer.constant):
             return self._lifter.lift(function.symbol)
 
-        return self.lift_const_addr(view, pointer)
-
-  
-    def lift_const_addr(self, view: BinaryView, pointer : mediumlevelil.MediumLevelILConstPtr):
-        """Lift a raw address:
-            - lift as char* if there is a string, otherwise as void* with raw bytes (if the datavariable was a ptr, lift as &ptr*, otherwise as ptr*)
-            - there were symbols which call them self recursively, therefore a small check before the end
-        """
-        variable = view.get_data_var_at(pointer.constant)
-        symbol = view.get_symbol_at(pointer.constant)
-
-        var_ref_string = (view.get_string_at(variable.value, True) or view.get_ascii_string_at(variable.value, min_length=2)) if variable and variable.value else None
-        var_ref_value = view.get_data_var_at(variable.value) if variable and variable.value else None
-
-        if var_ref_value and pointer.constant == var_ref_value.address: # Recursive ptr to itself (0x4040 := 0x4040), lift symbol if there, else just make a data_addr symbol
-            data_symbol =  view.get_symbol_at(variable.value)
-            var_ref_value = data_symbol if data_symbol else Symbol("data_" + f"{pointer.constant:x}", pointer.constant, vartype=Integer.uint32_t())    
-
-        g_var = GlobalVariable(
-            name=symbol.name[:-2] if symbol and symbol.name.find(".0") != -1 else symbol.name if symbol else "data_" + f"{pointer.constant:x}",
-            vartype=self._lifter.lift(Type.pointer(view.arch, Type.char())) if var_ref_string else self._lifter.lift(Type.pointer(view.arch, Type.void())),
-            ssa_label=pointer.ssa_memory_version if pointer else 0,
-            initial_value=self._lifter.lift(var_ref_value, view=view, parent=pointer) if var_ref_value else Constant(var_ref_string.value) \
-            if var_ref_string else self._get_raw_bytes(view, pointer.constant)
-        ) 
+        variable = DataVariable(view, pointer.constant, Type.void(), False)
+        global_variable = self._lifter.lift(variable, view=view, parent=pointer)
 
-        return UnaryOperation(OperationType.address,[g_var]) if variable is not None and isinstance(variable.type, PointerType) else g_var
+        return self._propagate_global_string(global_variable, variable, view)
 
+    def _propagate_global_string(self, globalVariable: GlobalVariable, variable: DataVariable, view: BinaryView) -> StringSymbol:
```

Comment:
Instead of having a symbol, the value of the symbol is placed in the code. 
```c
const char* str = "Hello World" 
printf("%s", str);

```
vs. 

```c
printf("%ls", "Hello World");
```

Renamed to `_replace_global_variable_with_value`. 
Better? Or too long?

Reply:
Nope, it is much better :+1: